mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1248,6 +1249,53 @@ impl TakerSwap {
     }
 
     async fn wait_for_taker_payment_spend(&self) -> Result<(Option<TakerSwapCommand>, Vec<TakerSwapEvent>), String> {
+        let mut watcher_abort_handle = None;
+        if self.ctx.use_watchers() {
```

Comment:
This should be done right after taker payment is sent. For example, if a taker sends payment and goes offline for some reason, they might not reach `wait_for_taker_payment_spend` stage, so watchers won't be aware about their transaction. The maker on their side will be aware of it, so they will proceed with swap process as usual.

Reply:
Moved it to `send_taker_payment` method. But now the watcher message is broadcast only once after the taker payment is sent. Is it enough? If not, I can also additionally start a broadcast loop with `broadcast_swap_message_every` in the `wait_for_taker_payment_spend` method.

Reply:
Yes, let's re-broadcast it periodically, too. If sent only once, the message might never reach other nodes due to occasional network outage.